### PR TITLE
runtime-rs: bring hybrid vsock devices in manager.

### PR DESCRIPTION
**Pull request by Apokleos**
Currently, virtio_vsock are still outside of the device manager. 
This causes some management issues,such as the inability to unify PCI address management.

Just do some work for hybrid vsock.

Fixes: #7655

### CODE REVIEW

File: src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs [hunk 1]
```diff
@@ -101,6 +102,15 @@ impl VirtSandbox {
     ) -> Result<Vec<ResourceConfig>> {
         let mut resource_configs = vec![];
 
+        // prepare VM hybrid vsock device config
+        // Add the hybrid vsock device first.
+        info!(sl!(), "prepare hybrid vsock resource for sandbox.");
+        let vm_hvsock = ResourceConfig::HybridVsock(HybridVsockConfig {
+            guest_cid: 3,
```
Comment (Ji-Xinyou) [inline src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs:109]: Why we are always using `cid=3`?

File: src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs [hunk 2]
```diff
@@ -127,6 +128,9 @@ impl CloudHypervisorInner {
     }
 
     async fn handle_hvsock_device(&mut self, _cfg: &HybridVsockConfig) -> Result<()> {
+        // uds path is to be used for adding hvsock device in VMM.
+        let _uds_path = [&self.jailer_root, DEFAULT_HYBRID_VSOCK_NAME].join("/");
```
Comment (Ji-Xinyou) [inline src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs:132]: If the hvsock logic is in the device manager entirely, we can simply return Ok(()) here.

File: src/runtime-rs/crates/hypervisor/src/device/driver/virtio_vsock.rs [hunk 3]
```diff
@@ -27,6 +34,47 @@ pub struct HybridVsockDevice {
...
+        // no need to do detach, just return Ok(None)
+        Ok(None)
+    }
+
+    async fn get_device_info(&self) -> DeviceType {
+        DeviceType::HybridVsock(self.clone())
+    }
+
+    async fn increase_attach_count(&mut self) -> Result<bool> {
+        // hybrivsock devices will not be attached multiple times, Just return Ok(false)
```
Comment (Ji-Xinyou) [inline src/runtime-rs/crates/hypervisor/src/device/driver/virtio_vsock.rs:66]: typo

**Review by Ji-Xinyou — COMMENTED** [review]
Some comments below, is this still WIP?

File: src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs (hunk 2 above)
Comment (Apokleos) [reply src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs]: fixed

File: src/runtime-rs/crates/hypervisor/src/device/driver/virtio_vsock.rs (hunk 3 above)
Comment (Apokleos) [reply src/runtime-rs/crates/hypervisor/src/device/driver/virtio_vsock.rs]: fixed

**Review by Tim-0731-Hzt — APPROVED** [review]
Thanks @Apokleos LGTM

**Review by quanweiZhou — APPROVED** [review]
Thanks @Apokleos, LGTM.

**Review by lifupan — APPROVED** [review]

**Pull request merged**